lib/puppet/provider/service/smf.rb

```diff
@@ -123,6 +123,14 @@ def service_states
     svcs("-H", "-o", "state,nstate", @resource[:name]).chomp.split
   end
 
+  def exists?
+    # non-zero exit status is a non-existent service
+    Puppet::Util::Execution.execute(
+      [command(:svccfg), "select", @resource[:name]],
+      :failonfail => false, :combine => true
+    )[1] == 0
+  end
+
```

Comment:
I think this resolves PUP-6860; it could be extracted into
```Ruby
svccfg ...
rescue
```
but I'm not sure it is better.